### PR TITLE
add depthwise_conv2d to paddle backend

### LINKED ISSUE #17440: depthwise_conv2d

### PR DESCRIPTION
**Pull request by mohame54**
close #17440

### CODE REVIEW

**Comment by mohame54** [comment]
@MahmoudAshraf97 I am ready for the review

File: ivy/functional/backends/paddle/layers.py [hunk 1]
```diff
@@ -139,6 +139,10 @@ def conv2d_transpose(
 
 
 # noinspection PyUnresolvedReferences
+@with_unsupported_device_and_dtypes(
+    {"2.4.2 and below": {"cpu": ("float16",)}},
+    backend_version,
+)
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/paddle/layers.py:145]:
```suggestion
@with_unsupported_device_and_dtypes(
    {"2.5.0 and below": {"cpu": ("float16",)}},
    backend_version,
)
```

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
please solve the failing tests and rerequest a review
```python
E           AssertionError:  the results from backend paddle and ground truth framework jax do not match
E            [[[[0.26092045]
E              [0.51092045]]]]!=[[[[0.59113104]
E              [0.25      ]]]] 
E           
E           
E           Falsifying example: test_depthwise_conv2d(
E               backend_fw=<module 'ivy.functional.backends.paddle' from '/workspaces/ivy/ivy/functional/backends/paddle/__init__.py'>,
E               on_device='cpu',
E               x_f_d_df=(['float64'], array([[[[0.68226207],
E                         [0.5218409 ],
E                         [0.5       ],
E                         [0.5       ]]]]), array([[[0.5]],
E                
E                       [[0.5]]]), [2, 3], 'NCHW', 2, 'SAME', 1),
E               test_flags=FunctionTestFlags(
E                   num_positional_args=4,
E                   with_out=False,
E                   instance_method=False,
E                   test_gradients=None,
E                   test_compile=None,
E                   as_variable=[False],
E                   native_arrays=[False],
E                   container=[False],
E               ),
E               ground_truth_backend='jax',
E               fn_name='depthwise_conv2d',
E           )
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.80.0', b'AXicY2BhZGYAAUZGRiYGRiAEggPsi17s0uvaBWJy8Iqxx5ZcADEZ4IAgEwIAL14KIA==') as a decorator on your test case
```

File: ivy/functional/backends/paddle/layers.py (hunk 1 above)
Comment (mohame54) [reply ivy/functional/backends/paddle/layers.py:145]: I don't know what's wrong with these tests

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hey @mohame54, thanks for looking into this! :smile: 

You can test if your code works using this - 
`pytest /workspaces/ivy/ivy_tests/test_ivy/test_functional/test_nn/test_layers.py::test_depthwise_conv2d`

I noticed something off with the output from running the pytest which I don't think is related to the changes you made, so dropped a note on our [Discord](https://discord.com/channels/799879767196958751/982738436383445073/1131487320588165230). 

I'll get back to your PR after that's looked into but in case I take longer than expected or if that pytest error is related to something else, feel free to request a review. Thanks! :smile:

**Pull request closed without merging**